src/main/java/org/sagebionetworks/bridge/udd/synapse/SynapseDownloadFromTableTask.java

```diff
@@ -0,0 +1,421 @@
+            // Copy headers.
+            modifiedCsvFileWriter.writeNext(csvFileReader.readNext());
+
+            // Iterate through the rows, replacing the file handle IDs with zip entry names.
+            String[] row;
+            while ((row = csvFileReader.readNext()) != null) {
+                // Clear health code.
+                row[healthCodeIdx] = null;
+
+                // Replace file handle IDs with zip entry names (if known)
+                for (int oneFileHandleColIdx : fileHandleColIdxSet) {
+                    String fileHandleId = row[oneFileHandleColIdx];
+                    if (Strings.isNullOrEmpty(fileHandleId)) {
+                        // blank column, skip
+                        continue;
+                    }
+
+                    String replacement = fileHandleIdToReplacement.get(fileHandleId);
+                    if (!Strings.isNullOrEmpty(replacement)) {
+                        row[oneFileHandleColIdx] = replacement;
+                    } else {
+                        row[oneFileHandleColIdx] = ERROR_DOWNLOADING_ATTACHMENT;
+                    }
+                }
+
+                // Write modified row to modifiedCsvFileWriter
+                modifiedCsvFileWriter.writeNext(row);
+            }
+        } catch (IOException ex) {
+            throw new AsyncTaskExecutionException("Error updating attachment file paths in file " +
+                    editedCsvFilePath + ": " + ex.getMessage(), ex);
+        } finally {
+            editCsvStopwatch.stop();
+            LOG.info("Updating attachment file paths in file " + editedCsvFilePath + " took " +
+                    editCsvStopwatch.elapsed(TimeUnit.MILLISECONDS) + " ms");
+        }
+
+        // rename editedCsvFile into csvFile, replacing the original csvFile
+        try {
+            fileHelper.moveFiles(editedCsvFile, ctx.getCsvFile());
+        } catch (IOException ex) {
+            throw new AsyncTaskExecutionException("Error moving (replacing) file from " + editedCsvFilePath +
+                    " to " + ctx.getCsvFilePath() + ": " + ex.getMessage(), ex);
+        }
+    }
+
+    /**
+     * <p>
+     * This is called when an error is thrown or if there's no data to download. We'll need to delete all intermediate
+     * files to ensure we leave the file system in the state we started it in. The specific intemediate files in
+     * question are {@link SynapseDownloadFromTableContext#getCsvFile},
+     * {@link SynapseDownloadFromTableContext#getBulkDownloadFile},
+     * {@link SynapseDownloadFromTableContext#getEditedCsvFile}, if any/all exist.
+     * </p>
+     * <p>
+     * This is package-scoped to enable unit tests.
+     * </p>
+     */
+    void cleanupFiles() {
+        List<File> filesToDelete = new ArrayList<>();
+        filesToDelete.add(ctx.getCsvFile());
+        filesToDelete.add(ctx.getBulkDownloadFile());
+        filesToDelete.add(ctx.getEditedCsvFile());
```

Comment:
I feel like this is easier to test if there's a single method that knows all the files it needs to clean up, rather than baking this logic into the catch block of a larger method.